crypto/hacl_lib.h

```diff
@@ -0,0 +1,231 @@
+/* 
+ * Copyright (c) 2016-2022 INRIA, CMU and Microsoft Corporation
+ * Copyright (c) 2022-2023 HACL* Contributors
+ * Copyright (c) 2023 Cryspen
+ */
+
+#ifndef __Hacl_LIB_H
+#define __Hacl_LIB_H
+
+#include <asm/unaligned.h>
+#include <crypto/sha256_base.h>
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/string.h>
+
+typedef u128 FStar_UInt128_uint128;
+
+static inline u128 FStar_UInt128_shift_left(u128 x, u32 y)
+{
+	return (x << y);
+}
+
+static inline u128 FStar_UInt128_add(u128 x, u128 y)
+{
+	return (x + y);
+}
+
+static inline u128 FStar_UInt128_uint64_to_uint128(u64 x)
+{
+	return ((u128)x);
+}
+
+/* Loads and stores. These avoid undefined behavior due to unaligned memory
+ * accesses, via memcpy. */
+
+#define load32_be(b) (get_unaligned_be32(b))
+#define store32_be(b, i) put_unaligned_be32(i, b);
+#define load64_be(b) (get_unaligned_be64(b))
+#define store64_be(b, i) put_unaligned_be64(i, b);
+
+static inline void store128_be(u8 *buf, u128 x)
+{
+	store64_be(buf, (u64)(x >> 64));
+	store64_be(buf + 8, (u64)(x));
+}
+
+/* Macros for prettier unrolling of loops */
+#define KRML_LOOP1(i, n, x) \
+	{                   \
+		x i += n;   \
+	}
```

Comment:
```suggestion
#define KRML_LOOP1(i, n, x) \
	do {                    \
		x i += n;           \
	} while(0)
```

Also, it seems dangerous to pass `if(0)` in the body of the loop.
Does wrapping `{x}` can help?